tests/cbmc/proofs/Makefile.common

```diff
@@ -0,0 +1,279 @@
+
+# Here, whenever there is a change in any of ANSI-C source
+# dependency files, make will take action. However, to make
+# sure changes in the headers files will also trigger make,
+# the user must run make clean first.
+$(ENTRY_GOTO)0.goto: setup_dependencies $(ENTRY).c $(DEPENDENCIES)
+	mkdir -p $(dir $@)
+	mkdir -p $(dir $(call LOG_FROM_ENTRY,$@))
+	$(call DO_GOTO_CC,\
+		--export-function-local-symbols \
+			$(CBMC_VERBOSITY) \
+			--function $(ENTRY) \
+			$(INC) \
+			$(DEFINES), \
+		$(ENTRY).c $(DEPENDENCIES),\
+		$@)
+
+# Removes specified function bodies. This allows us to replace
+# function definitions with ABSTRACTIONS.
+$(ENTRY_GOTO)1.goto: $(ENTRY_GOTO)0.goto
+ifeq ($(REMOVE_FUNCTION_BODY),"")
+	$(call DO_NOOP_COPY,"Not removing function bodies",$<,$@)
+else
+	$(call DO_GOTO_INSTRUMENT,$(REMOVE_FUNCTION_BODY),$<,$@)
+endif
+
+# ABSTRACTIONS is a list of function stubs to use. If a function body
+# is missing and is not abstracted, then it returnes a non
+# deterministic value.
+$(ENTRY_GOTO)2.goto: $(ENTRY_GOTO)1.goto
+ifeq ($(ABSTRACTIONS),"")
+	$(call DO_NOOP_COPY,"Not implementing abstractions",$<,$@)
+else
+	$(call DO_GOTO_CC,--function $(ENTRY) $(ABSTRACTIONS) $(INC) $(DEFINES),$<,$@)
+endif
+
+# Simplify and constant propagation may benefit from unwinding first
+$(ENTRY_GOTO)3.goto: $(ENTRY_GOTO)2.goto
+ifeq ($(UNWIND_GOTO),1)
+	$(call DO_GOTO_INSTRUMENT,$(UNWINDING),$<,$@)
+else
+	$(call DO_NOOP_COPY,"Not unwinding goto program",$<,$@)
+endif
+
+# Skip simplify (and hence generate-function-body) until missing source locations debugged
+$(ENTRY_GOTO)4.goto: $(ENTRY_GOTO)3.goto
+ifeq ($(SIMPLIFY),1)
+	$(call DO_GOTO_INSTRUMENT,--generate-function-body '.*',$<,$@)
+else
+	$(call DO_NOOP_COPY,"Not generating-function-bodies in goto program",$<,$@)
+endif
+
+# Skip simplify (and hence generate-function-body) until missing source locations debugged
+$(ENTRY_GOTO)5.goto: $(ENTRY_GOTO)4.goto
+ifeq ($(SIMPLIFY),1)
+	$(call DO_GOTO_ANALYZER,--simplify,$@,$<)
+else
+	$(call DO_NOOP_COPY,"Not simplfying goto program",$<,$@)
+endif
+
+# Simplify the goto program by removing any unused function bodies
+$(ENTRY_GOTO)6.goto: $(ENTRY_GOTO)5.goto
+	$(call DO_GOTO_INSTRUMENT,--drop-unused-functions,$<,$@)
+
+# Simplify the goto program by slicing away initializations of unused
+# global variables
+$(ENTRY_GOTO)7.goto: $(ENTRY_GOTO)6.goto
+	$(call DO_GOTO_INSTRUMENT,--slice-global-inits,$<,$@)
+
+$(ENTRY_GOTO).goto: $(ENTRY_GOTO)7.goto
+	cp $< $@
+
+goto: $(ENTRY_GOTO).goto
+	echo $(DEPENDENT_GOTOS)
+
+################################################################
+# Targets to run the analysis commands
+logs/cbmc.log: $(ENTRY_GOTO).goto
+	cbmc $(CBMCFLAGS) --trace $< 2>&1 | tee $@
+
+logs/property.xml: $(ENTRY_GOTO).goto
+	cbmc $(CBMCFLAGS) --show-properties --xml-ui $< 2>&1 > $@
+
+logs/coverage.xml: $(ENTRY_GOTO).goto
+	cbmc $(filter-out --unwinding-assertions,$(CBMCFLAGS)) --cover location --xml-ui $< 2>&1 > $@
```

Comment:
coverage and property will fail the whole build if they fail, but cbmc.log won't (because it's being piped into `tee` which always returns zero). Is that okay? My feeling is that you should be piping all three of these commands into `tee` (or otherwise ignoring their return codes)

Reply:
Hmm, why is it not using `DO_AND_LOG_COMMAND`, which would make sure that the failure is not silent?

Reply:
Done